lib/widgets/currency_converter_dialog.dart

```diff
@@ -0,0 +1,255 @@
+import 'package:breez/bloc/account/account_model.dart';
+import 'package:breez/bloc/account/fiat_conversion.dart';
+import 'package:breez/bloc/blocs_provider.dart';
+import 'package:breez/bloc/user_profile/currency.dart';
+import 'package:breez/bloc/user_profile/user_profile_bloc.dart';
+import 'package:breez/theme_data.dart' as theme;
+import 'package:breez/widgets/loader.dart';
+import 'package:fixnum/fixnum.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+
+class CurrencyConverterDialog extends StatefulWidget {
+  final Function(String string) _onConvert;
+  final String Function(Int64 amount) validatorFn;
+
+  CurrencyConverterDialog(this._onConvert, this.validatorFn);
+
+  @override
+  CurrencyConverterDialogState createState() {
+    return new CurrencyConverterDialogState();
+  }
+}
+
+class CurrencyConverterDialogState extends State<CurrencyConverterDialog> with SingleTickerProviderStateMixin {
+  final _formKey = GlobalKey<FormState>();
+  final TextEditingController _fiatAmountController = new TextEditingController();
+  final FocusNode _fiatAmountFocusNode = FocusNode();
+
+  AnimationController _controller;
+  Animation<Color> _colorAnimation;
+
+  AccountBloc _accountBloc;
+  UserProfileBloc _userProfileBloc;
+
+  Currency _currency = Currency.BTC;
+  FiatConversion _fiatCurrency;
+  List<FiatConversion> _fiatConversionList = List<FiatConversion>();
+
+  Int64 _satoshies;
+  double _exchangeRate;
+
+  bool _isInit = false;
+
+  @override
+  void initState() {
+    super.initState();
+    _fiatAmountController.addListener(_convertCurrency);
+
+    _controller = AnimationController(vsync: this, duration: Duration(milliseconds: 400));
+    _colorAnimation = new ColorTween(
+      begin: Colors.black45,
+      end: theme.BreezColors.blue[500],
+    ).animate(_controller)
+      ..addListener(() {
+        setState(() {});
+      });
+    // Loop back to start and stop
+    _controller.addStatusListener((status) {
+      if (status == AnimationStatus.completed) {
+        _controller.reverse();
+      } else if (status == AnimationStatus.dismissed) {
+        _controller.stop();
+      }
+    });
+  }
+
+  @override
+  void didChangeDependencies() {
+    if (!_isInit) {
+      _accountBloc = AppBlocsProvider.of<AccountBloc>(context);
+      _userProfileBloc = AppBlocsProvider.of<UserProfileBloc>(context);
+      _isInit = true;
+    }
+    super.didChangeDependencies();
+  }
+
+  @override
+  void dispose() {
+    _fiatAmountController.dispose();
+    _controller?.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return StreamBuilder<AccountModel>(
+        stream: _accountBloc.accountStream,
+        builder: (context, snapshot) {
+          AccountModel account = snapshot.data;
+          if (!snapshot.hasData) {
+            return Container();
+          }
+
+          _currency = account.currency;
+          _fiatCurrency = account.fiatCurrency;
```

Comment:
Don't need this variable _fiatCurrency, everything can be taken from accountModel.
Even if it is needed on state, it is not a good practice to initialize it within the build method.